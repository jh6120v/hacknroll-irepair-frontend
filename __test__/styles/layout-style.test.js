import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import {
    Container, Header, Logo, Wrapper
} from '../../src/styles/layout-style';

describe('test layout-style.', () => {
    it('test <Header /> snapshot.', () => {
        const { container } = render(<Header />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <Header /> when render it.', () => {
        const { container } = render(<Header />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('justify-content', 'center');
        expect(tree).toHaveStyleRule('height', '34vh');
        expect(tree).toHaveStyleRule('padding', '11vh 0 5vh 0');
    });

    it('test <Logo /> when render it.', () => {
        const { container } = render(<Logo />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('width', '100%');
        expect(tree).toHaveStyleRule('background-image', 'url(i-repair-logo.svg)');
        expect(tree).toHaveStyleRule('background-position', 'center center');
        expect(tree).toHaveStyleRule('background-size', 'contain');
        expect(tree).toHaveStyleRule('background-repeat', 'no-repeat');
    });

    it('test <Wrapper /> snapshot.', () => {
        const { container } = render(<Wrapper />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <Wrapper /> when render it.', () => {
        const { container } = render(<Wrapper />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('max-width', '400px');
        expect(tree).toHaveStyleRule('min-height', '100%');
        expect(tree).toHaveStyleRule('flex-wrap', 'wrap');
        expect(tree).toHaveStyleRule('flex-direction', 'column');
        expect(tree).toHaveStyleRule('margin', '0');
        expect(tree).toHaveStyleRule('margin', '0 auto', {
            media: '(min-width:480px)'
        });
    });

    it('test <Container /> snapshot.', () => {
        const { container } = render(<Container />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <Container /> when render it.', () => {
        const { container } = render(<Container />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('width', '100%');
        expect(tree).toHaveStyleRule('height', '100%');
        expect(tree).toHaveStyleRule('flex-grow', '1');
        expect(tree).toHaveStyleRule('flex-shrink', '1');
        expect(tree).toHaveStyleRule('flex-basis', 'auto');
    });
});
