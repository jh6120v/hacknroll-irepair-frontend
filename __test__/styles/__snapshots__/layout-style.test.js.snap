// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test layout-style. test <Container /> snapshot. 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
}

<div
  class="c0"
/>
`;

exports[`test layout-style. test <Header /> snapshot. 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 34vh;
  padding: 11vh 0 5vh 0;
}

<div
  class="c0"
/>
`;

exports[`test layout-style. test <Wrapper /> snapshot. 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: 400px;
  min-height: 100%;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0;
}

@media (min-width:480px) {
  .c0 {
    margin: 0 auto;
  }
}

<div
  class="c0"
/>
`;
